modules/compatibility/src/test/java/org/apache/ignite/compatibility/testframework/junits/IgniteCompatibilityAbstractTest.java

```diff
@@ -0,0 +1,331 @@
+    public static final String SYNCHRONIZATION_LOG_MESSAGE_JOINED = "Remote node has joined, id: ";
+
+    /** Using for synchronization of nodes startup in case of starting remote nodes first. */
+    public static final String SYNCHRONIZATION_LOG_MESSAGE_PREPARED = "Remote node has prepared, id: ";
+
+    /** Waiting seconds of the join of a node to topology. */
+    protected static final int NODE_JOIN_TIMEOUT = 30;
+
+    /** Local JVM Ignite node. */
+    protected Ignite locJvmInstance = null;
+
+    /** Remote JVM Ignite instance. */
+    protected Ignite rmJvmInstance = null;
+
+    /**
+     * Gets a path to the default DB working directory.
+     *
+     * @return Path to the default DB working directory.
+     * @throws IgniteCheckedException In case of an error.
+     * @see #deleteDefaultDBWorkDirectory()
+     * @see #defaultDBWorkDirectoryIsEmpty()
+     */
+    protected Path getDefaultDbWorkPath() throws IgniteCheckedException {
+        return Paths.get(U.defaultWorkDirectory() + File.separator + "db");
+    }
+
+    /**
+     * Deletes the default DB working directory with all sub-directories and files.
+     *
+     * @return {@code true} if and only if the file or directory is successfully deleted, otherwise {@code false}.
+     * @throws IgniteCheckedException In case of an error.
+     * @see #getDefaultDbWorkPath()
+     * @see #deleteDefaultDBWorkDirectory()
+     */
+    protected boolean deleteDefaultDBWorkDirectory() throws IgniteCheckedException {
+        Path dir = getDefaultDbWorkPath();
+
+        return Files.notExists(dir) || U.delete(dir.toFile());
+    }
+
+    /**
+     * Checks if the default DB working directory is empty.
+     *
+     * @return {@code true} if the default DB working directory is empty or doesn't exist, otherwise {@code false}.
+     * @throws IgniteCheckedException In case of an error.
+     * @see #getDefaultDbWorkPath()
+     * @see #deleteDefaultDBWorkDirectory()
+     */
+    @SuppressWarnings("ConstantConditions")
+    protected boolean defaultDBWorkDirectoryIsEmpty() throws IgniteCheckedException {
+        File dir = getDefaultDbWorkPath().toFile();
+
+        return !dir.exists() || (dir.isDirectory() && dir.list().length == 0);
+    }
+
+    /** {@inheritDoc} */
+    @Override protected boolean isMultiJvm() {
+        return true;
+    }
+
+    /**
+     * Starts new Ignite instance of given version and index <b>in separate JVM</b>.
+     *
+     * Uses an ignite-core artifact in the Maven local repository, if it isn't exists there, it will be downloaded and
+     * stored via Maven.
+     *
+     * @param idx Index of the grid to start.
+     * @param ver Ignite version.
+     * @param cfgClos IgniteInClosure for post-configuration.
+     * @return Started grid.
+     * @throws Exception If failed.
+     */
+    protected IgniteEx startGrid(int idx, String ver, IgniteInClosure<IgniteConfiguration> cfgClos) throws Exception {
```

Comment:
Can we use some sort of enum for `ver` variable?

Reply:
I thought about it.
Not sure is it required.